framework/random.cpp

```diff
@@ -478,23 +500,26 @@ void random_init_global(const char *seed_from_user)
         sApp->random_engine->reloadGlobalState(ptr);
     } else {
         // it was a file, read our seed from there
-        uint8_t type;
-        if (read(fd, &type, 1) != 1) {
-            // will never happen
-            perror("read");
-            exit(EX_IOERR);
+        auto read_from_seed = [&](auto &where) {
+            off_t n = read(fd, &where, sizeof(where));
+            if (n != sizeof(where)) {
+                if (seed_from_user)
+                    fprintf(stderr, "%s: could not read from %s: %s\n", program_invocation_name,
+                            seed_from_user, n < 0 ? strerror(errno) : "File too short");
+                exit(EX_IOERR);
+            }
+            return true;
+        };
+
+        EngineType engine_type = LCG;
+        if (haveAes()) {
+            if (uint8_t type; read_from_seed(type) && (type & 1))
```

Comment:
Here I did change the interpretation. Previously, if bit 1 was set, we'd select the AES engine; now we do if bit 1 is set.

Updating.